File: backend/app/services/utils/player.py
from app.types.race import PlayerType, Race


OwnrPlayerTypeDict: dict[int, PlayerType] = {
  0: "Inactive",
  1: "Computer (game)",
  2: "Occupied By Human Player",
  3: "Rescue",
  4: "Unused",
  5: "Computer",
  6: "Human (Open Slot)",
  7: "Neutral",
  8: "Closed Slot",
}

SidePlayerRaceDict: dict[int, Race] = {
  0: "Zerg",
  1: "Terran",
  2: "Protoss",
  3: "Invalid (Independant)",
  4: "Invalid (Neutral)",
  5: "User Selectable",
  6: "Random",
  7: "Inactive",
}

SidePlayerRaceReverseDict: dict[Race, int] = {
  v: k for k, v in SidePlayerRaceDict.items()
}

OwnrPlayerTypeReverseDict: dict[PlayerType, int] = {
  v: k for k, v in OwnrPlayerTypeDict.items()
}

File: frontend/types/schemas/unit/Player.ts
import { z } from "zod";

export const PlayerSchema = z.object({
  id: z.number().int().default(0),
  name: z.string().default("Object"),
  color: z.number().int(),
  rgb_color: z.array(z.any()).min(3).max(3),
  player_type: z.enum([
    "Inactive",
    "Computer (game)",
    "Occupied By Human Player",
    "Rescue",
    "Unused",
    "Computer",
    "Human (Open Slot)",
    "Neutral",
    "Closed Slot",
  ]),
  race: z.enum([
    "Zerg",
    "Terran",
    "Protoss",
    "Invalid (Independant)",
    "Invalid (Neutral)",
    "User Selectable",
    "Random",
    "Inactive",
  ]),
  force: z.number().int().gte(0).lt(4).default(0),
});
export type Player = z.infer<typeof PlayerSchema>;
